### ISSUE TITLE
Mutually exclusive backends? CUDA Clang + OpenMP

### ISSUE THREAD

**Issue by jjwilke** [kokkos/kokkos#2250]
Trying random combinations of configure options. If you compile for both OpenMP and CUDA backends with clang, the compilation crashes:

````
/ascldap/users/jjwilke/Programs/kokkos/cmake/core/src/OpenMP/Kokkos_OpenMP_Exec.hpp:67:10: fatal error: 'omp.h' file not found
#include <omp.h>
````
Does someone know more about this? It looks like Clang OpenMP might not support openmp gpu offload. Possibly there are more combinations like this that are not allowed.

**Comment by jjwilke** [comment]
Okay. Confirmed that you can include `omp.h` even if you don't have the `__CUDA__` header guards. Really the issue here is that just knowing the clang version doesn't actually tell you that you have OpenMP. We probably want a feature test for clang to ensure that openmp was actually built as a component.

**Comment by masterleinad** [comment]
Addressed in #2337.
